### PR TITLE
Restructuring the B2B components space

### PR DESCRIPTION
**Pull request by sushmangupta**
I have restructured the B2B components space and also added the contents of this PR here : https://github.com/shopware/docs/pull/1396

### CODE REVIEW

File: products/extensions/b2b-components/index.md [hunk 1]
```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
 * **Quick Order and Shopping List** takes care of distinctive B2B buying behaviors.
 
 * **Digital Sales Composables** aims to provide a set of composable frontends to cover more complex Sales Representative jobs.
+
+## Configuring custom toggles for B2B components
+
+The B2B Components allow merchants to selectively choose and configure B2B capabilities according to their needs. They offer merchants the ability to craft a tailored B2B ecommerce experience for their business partners while also allowing agencies to fine-tune Shopware to meet specific requirements. This means that B2B Components can be individually activated or deactivated for each business partner within the shop.
```
Comment (sushmangupta) [inline products/extensions/b2b-components/index.md:26]: There are lot of synonyms used. Do we call it as B2B Components / B2B features / B2B Capabilities across the docs ?

File: products/extensions/b2b-components/index.md [hunk 2]
```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
...
+
+The following articles will guide you how to do this by creating custom toggles via a plugin for B2B Components (Customer-specific features).
+
+The **Customer-specific features** section on the Customer detail page allows the shop merchant to turn these B2B features on or off for a specific customer. 
+
+![Feature Toggles](../../../assets/b2b-feature-toggles.png)
+
+To achieve this, ACL and address the following cases where functionality may be hidden:
```
Comment (sushmangupta) [inline products/extensions/b2b-components/index.md:34]: May I know what this sentence mean ?

File: products/extensions/b2b-components/index.md [hunk 3]
```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
...
+### Prerequisite
+
+To improve organization and maintain a clear structure, it is advisable to relocate all B2B Components into the `B2B` folder within the Commercial plugin. By doing so, you can centralize the B2B-related functionality, making it easier to locate, manage, and maintain the codebase. This folder structure promotes better separation of concerns and enhances the overall modularity of the application.
+
+```
+├── src
+│   ├── B2B
+│   │   ├── QuickOrder
+│   │   ├── AnotherB2BComponent
+│   │   │   CommercialB2BBundle.php
```
Comment (sushmangupta) [inline products/extensions/b2b-components/index.md:50]: where is this "CommercialB2BBundle.PHP" fall. Is it under "AnotherB2BComponent" or elsewhere?
If its under "AnotherB2BComponent", then we need to mark a line there.

File: products/extensions/b2b-components/index.md [hunk 4]
```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
...
+                'type' => self::TYPE_B2B,
+            ],
+        ];
+    }
+}
+```
+
+## Using feature toggle in Route/API/Controller
+
+### Using new service
```
Comment (sushmangupta) [inline products/extensions/b2b-components/index.md:80]: Is this heading necessary to be mentioned ?

File: products/extensions/b2b-components/index.md [hunk 5]
```diff
@@ -20,3 +20,134 @@ In the world of digital B2B commerce, where businesses engage with other compani
...
+            throw CustomerSpecificFeatureException::notAllowed('QUICK_ORDER');
+        }
+
+        ...
+    }
+```
+
+## Using feature toggle in Twig - Storefront
+
+### New Twig function
```
Comment (sushmangupta) [inline products/extensions/b2b-components/index.md:113]: Is this heading necessary to be mentioned ?

File: products/extensions/b2b-components/shopping-lists/guides/entities-and-schema.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+## Entities
+
+### Shopping Lists
+
+Shopping lists represent a list of products prepared for a customer or a sales channel. They show basic information about the product list, such as the name, customer, sales channel and so on.
+
+### Line item
+
+The Shopping lists line item represents a set of products. These products are line items of the shopping list.
```
Comment (sushmangupta) [inline products/extensions/b2b-components/shopping-lists/guides/entities-and-schema.md:18]: This sentence seems to be quite unclear / overlapping with its meaning. Can it be rephrased ?

File: products/extensions/b2b-components/index.md (hunk 1 above)
Comment (nam4am) [reply products/extensions/b2b-components/index.md:26]: Mostly I only see using B2B components for all relevant cases. As a way to differentiate from B2B suite. Sometimes we use B2B features but only for discuss technical purposes.

Therefore, in my opinion, we should only use B2B components for consistency

File: products/extensions/b2b-components/index.md (hunk 3 above)
Comment (nam4am) [reply products/extensions/b2b-components/index.md:50]: It is the only file here, so it doesn't fall under "AnotherB2BComponent" or anywhere else.

File: products/extensions/b2b-components/index.md (hunk 2 above)
Comment (nam4am) [reply products/extensions/b2b-components/index.md:34]: I believe we need to clarify two points: 

The first one is Customer-specific features: Enable Merchants to show **specific customers** using B2B components. 

The second point is the ACL, which enables merchants to define permissions for **shop users** to Read, Update, Delete, Create, with specific B2B components.

File: products/extensions/b2b-components/index.md (hunk 4 above)
Comment (nam4am) [reply products/extensions/b2b-components/index.md:80]: We can remove it, due the first sentences we said "You can use a new Twig extension called..."

File: products/extensions/b2b-components/index.md (hunk 5 above)
Comment (nam4am) [reply products/extensions/b2b-components/index.md:113]: We can remove it, due the first sentences we said "You can use a new Twig extension called..."

File: products/extensions/b2b-components/shopping-lists/guides/entities-and-schema.md (hunk 6 above)
Comment (leduc92) [reply products/extensions/b2b-components/shopping-lists/guides/entities-and-schema.md:18]: Maybe: The Shopping List Line Item represents individual products within a shopping list. Each product in the shopping list is considered a line item.

File: products/extensions/b2b-components/employee-management/guides/b2b-roles.md [hunk 7]
```diff
@@ -9,4 +9,4 @@ nav:
 
 Roles can be used to bind multiple permission to employees with contexts. Every employee can have one assigned role. Based on that role and the containing permission, the employee will get access to certain information and functionalities.
 
-The business partner has the opportunity to create a “default” role that will be selected by default, when creating a new employee.
+The business partner has the opportunity to create a **default** role that will be selected by default, when creating a new employee.
```
Comment (sushmangupta) [inline products/extensions/b2b-components/employee-management/guides/b2b-roles.md:12]: Is this necessary as its already described here : https://developer.shopware.com/docs/products/extensions/b2b-components/employee-management/concept/entities-and-schema.html#role

Comment (nam4am) [reply products/extensions/b2b-components/employee-management/guides/b2b-roles.md:12]: I think it still necessary, the role description in section entities-and-schema used for explain the meaning for table/entity swag_b2b_role in database. And the role in B2B-roles, explain more detail about how Role business logic work.

**Review by bojanrajh — APPROVED** [review]

**Pull request merged**